coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelNodeToCoralSqlNodeTransformer.java

```diff
@@ -0,0 +1,172 @@
+
+  /**
+   * Creates a CoralRelNodeToCoralSqlNodeTransformer.
+   */
+  CoralRelNodeToCoralSqlNodeTransformer() {
+    super(INSTANCE);
+  }
+
+  private static SqlDialect returnInstance() {
+    SqlDialect.Context context = SqlDialect.EMPTY_CONTEXT.withDatabaseProduct(SqlDialect.DatabaseProduct.HIVE)
+        .withNullCollation(NullCollation.HIGH);
+
+    return new SqlDialect(context);
+  }
+
+  /**
+   * This overridden function makes sure that the basetable names in the output SqlNode
+   * will be in the format - "dbname.tablename" instead of "catalogname.dbname.tablename"
+   * For Example:
+   * hive.default.complex will be converted to default.complex
+   *
+   * input RelNode format is:
+   * LogicalTableScan(table=[[hive, default, complex]])
+   * This overriding function creates a SqlIdentifier(default, complex)
+   * instead of super's default: SqlIdentifier(hive, default, complex)
+   */
+  @Override
+  public Result visit(TableScan e) {
+    //todo: look into removing this transformation during hiveToRelConversion
+    checkQualifiedName(e.getTable().getQualifiedName());
+    List<String> tableNameWithoutCatalogAndDb = e.getTable().getQualifiedName().subList(1, 3);
+    final SqlIdentifier identifier = new SqlIdentifier(tableNameWithoutCatalogAndDb, SqlParserPos.ZERO);
+    return result(identifier, ImmutableList.of(Clause.FROM), e, null);
+  }
+
+  /**
+   * Correlate represents the FROM clause of the final sql query
+   * when the FROM clause contains explode() UDF as well
+   *
+   * Overriding the default implementation is required because
+   * the rightChildResult is also overriden and doesn't match super's expectation
+   */
+  @Override
+  public Result visit(Correlate e) {
+    //Left Result removes the catalog name "hive" from the table path
+    final Result leftResult = visitChild(0, e.getLeft());
+
+    // Add context specifying correlationId has same context as its left child
+    correlTableMap.put(e.getCorrelationId(), leftResult.qualifiedContext());
+
+    final Result rightResult = visitChild(1, e.getRight());
+
+    SqlJoin join = new SqlJoin(POS, leftResult.asFrom(), SqlLiteral.createBoolean(false, POS),
+        JoinType.COMMA.symbol(POS), rightResult.asFrom(), JoinConditionType.NONE.symbol(POS), null);
+
+    SqlIdentifier leftNode = (SqlIdentifier) leftResult.node;
+    SqlBasicCall lateralRightNode = (SqlBasicCall) rightResult.node;
+    SqlBasicCall asSqlNode = (SqlBasicCall) lateralRightNode.getOperands()[0];
+    SqlIdentifier asSqlNodeAlias = (SqlIdentifier) asSqlNode.getOperands()[1];
+
+    //The default implementation assumes that the rightLateralNode, for LateralSqlOperator,
+    // has a null alias. At the same time, it expects a non-null alias for the same node.
+    // At this point, we can either customize the rightLateralNode to pass the alias check
+    // or we can bypass the check and work around it. The latter appoach has been chosen.
```

Comment:
approach

Reply:
We are doing things a little differently in this PR. 

As per calcite's implementation: 
`visit(Uncollect e)` is visited first, which returns a `Result`. This `Result` has a sqlNode which only represents `UNNEST(complex.c) AS t0` and is returned to `visit(Correlate e)` as the `rightResult`. In `visit(Correlate e)`, we first add a `LATERAL` operator on top of the rightResult and then finally return a `SqlJoin` by joining the `leftResult` and this transformed `rightResult`. 

In the current implementation, I have used `visit(Uncollect e)` to return a sqlNode which represents `LATERAL UNNEST(`complex`.`c`) AS t0 (ccol)`. Also, I have used `visit(Correlate e)` to only do a "join" between the leftResult and the rightResult without any additional transformations. I was trying to go for an implementation where we don't need to modify the child results. 

Since I'm trying to add a SqlJoinOperator, I just wanted to call [this](https://github.com/linkedin/linkedin-calcite/blob/li-1.21.0/core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java#L398) calcite function. However, the `ImmutableMap.Builder<String, RelDataType> builder` is not successfully created. It fails assertions [here](https://github.com/linkedin/linkedin-calcite/blob/li-1.21.0/core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java#L416) due to [this line  ](https://github.com/linkedin/linkedin-calcite/blob/li-1.21.0/core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorUtil.java#L319). 

Hence, I used a workaround to construct the `builder` in  `visit(Correlate e)` and directly call [this line](https://github.com/linkedin/linkedin-calcite/blob/li-1.21.0/core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java#L404) instead.

I drew the inspiration for this implementation choice from our existing Converters - `SparkRelToSparkSqlConverter` and tweaked it a bit more. Let me know if you feel that we should follow the same patterns as the parent implementation.